src/chat/chat_panel.rs

```diff
@@ -506,8 +506,14 @@ impl WidgetMatchEvent for ChatPanel {
                             self.focus_on_prompt_input_pending = true;
                         }
                     }
-                    ChatEntityId::Agent(agent) => {
-                        store.chats.create_empty_chat_with_agent(agent);
+                    ChatEntityId::Agent(agent_id) => {
+                        let agent = store
+                            .chats
+                            .available_agents
+                            .get(&agent_id)
+                            .cloned()
+                            .unwrap_or_default();
```

Comment:
mmm i think it may be better to ensure we can't start a chat with an unrecognized agent, and panic or eprintln something here to indicate a dev mistake

Reply:
Answering on the next one